### PR TITLE
Goal abortion.

### PR DESCRIPTION
**Pull request by dorkamotorka**

### CODE REVIEW

**Comment by tp4348** [comment]
Either one or another abort method should be used(one should be commented out for testing purposes), due to variables(distRemaining, prevDistance).

**Comment by dorkamotorka** [comment]
Either one or another abort method should be used(one should be commented out for testing purposes), due to variables(distRemaining, prevDistance).

File: src/move_basic.cpp [hunk 1]
```diff
@@ -241,6 +245,11 @@ MoveBasic::MoveBasic(): tfBuffer(ros::Duration(3.0)),
     // Reverse distances for which rotation won't be performed
     nh.param<double>("reverse_without_turning_threshold",
                       reverseWithoutTurningThreshold, 0.5);
+    
+    // Timeout for distance to goal update
+    nh.param<double>("abort_timeout", abortTimeout, 5.0);
```
Comment (rohbotics) [inline src/move_basic.cpp:250]: If these are set to 0, then we should disable that type of abort mode.

File: src/move_basic.cpp [hunk 2]
```diff
@@ -725,6 +736,25 @@ bool MoveBasic::moveLinear(tf2::Transform& goalInDriving,
         remaining = goalInBase.getOrigin();
         double distRemaining = sqrt(linear.x() * linear.x() + linear.y() * linear.y());
 
+// 
+	if (distRemaining < prevDistance) {
+		double current = ros::Time::now().toSec();
+		if (current-last > abortTimeout) {
+			abortGoal("MoveBasic: Goal update timeout.");
```
Comment (rohbotics) [inline src/move_basic.cpp]: A more detailed abort message would be useful.

"No progress towards goal for longer than timeout."

File: src/move_basic.cpp [hunk 3]
```diff
@@ -725,6 +736,25 @@ bool MoveBasic::moveLinear(tf2::Transform& goalInDriving,
...
+		if (current-last > abortTimeout) {
+			abortGoal("MoveBasic: Goal update timeout.");		
+		}
+		prevDistance = distRemaining;
+		last = current;
+	}
+
+	if (distRemaining < prevDistance) {
+		if (prevDistance+distThreshold < distRemaining) {
+			abortGoal("MoveBasic: Goal distance threshold exceeded.");
```
Comment (rohbotics) [inline src/move_basic.cpp]: Same thing here:

"Detected that we are moving further from the goal, aborting."

**Review by rohbotics — APPROVED** [review]

**Pull request merged**